pkg/ephemeral-disk-utils/utils.go

```diff
@@ -29,14 +29,20 @@ import (
 // TODO this should be part of structs, instead of a global
 var DefaultOwnershipManager OwnershipManagerInterface = &OwnershipManager{user: "qemu"}
 
+func SetNonRootDefault() {
+	DefaultOwnershipManager = &nonOpManager{}
+}
+
```

Comment:
Picking up from the discussion here, https://github.com/kubevirt/kubevirt/pull/4922#discussion_r658990113

i said  "why do we need this separate interface for NonRoot? Can't we make SetFileOwnership a no op if the right permissions already exist? I don't think we'd need to run chown to do that."

and what I meant by that is, i understand this this noOpManager is meant to prevent the chown. My point was, rather than creating this noOp interface that silently does nothing, instead couldn't we look up who owns the file and only attempt to set it when the ownership isn't already what we want it to be?

The assumption here is that in the non-root case, the correct ownership already exists on the file, so by looking up the ownership on the file and seeing that it's already what we want, chown wouldn't get called.

Reply:
>The assumption here is that in the non-root case, the correct ownership already exists on the file, so by looking up the ownership on the file and seeing that it's already what we want, chown wouldn't get called.

I am not seeing immediate benefit but it`s certainly not hard to change...

Reply:
two benefits.

- one code path for root and non root
- if a permissions issue does arise later on for some reason, we'll see it fail here immediately because non-root won't have chown capabilities (this would have been ignored with the no op and probably appeared elsewhere in an less obvious way"

Reply:
The last commit is addressing it.

Reply:
@davidvossel 
It`s causing some issues.